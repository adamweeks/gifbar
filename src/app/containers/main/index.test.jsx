import React from 'react';
import renderer from 'react-test-renderer';

import Main from '.';

jest.mock(`../../giphy-search`, () => {
  return {
    fetchGiphyTrending: jest.fn(() => Promise.resolve())
  };
});

jest.mock(`electron-json-storage`, () => {
  return {
    get: jest.fn(() => ({items: []})),
    set: jest.fn()
  };
});

describe(`Main component`, () => {
  electron.remote.BrowserWindow = jest.fn();
  electron.remote.getGlobal = jest.fn(() => global);
  electron.ipcRenderer = {
    on: jest.fn()
  };
  const component = renderer.create(
    <Main
      info={`Image Info`}
    />
  );

  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });
});
