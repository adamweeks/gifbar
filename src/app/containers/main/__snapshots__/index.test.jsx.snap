// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders properly 1`] = `
<div
  className="mainContainer"
>
  <div
    className="searchBar"
  >
    <div
      className="searchBar"
    >
      <form>
        <input
          id="searchBar"
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Search for gifs"
          type="text"
          value=""
        />
        <button
          className="cancel"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </form>
    </div>
  </div>
  <div
    className="results"
  >
    <div>
      <p>
        Could not load trending gifs, giphy not found!
      </p>
      <p>
        <img
          src="https://media3.giphy.com/media/l3V0HLYPfIKIVDyBG/giphy.gif"
        />
      </p>
    </div>
  </div>
  <div
    className="attribution"
  >
    <div>
      <div
        className="footer"
      >
        <div
          className="side"
        >
          <div
            className="button"
            onClick={[Function]}
          >
            <img
              className="imgButton"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          className="main"
        >
          <div
            className="title"
          >
            Top Trending GIFs!
          </div>
          <div
            className="attribution"
          >
            <img
              className="img"
              src="test-file-stub"
            />
          </div>
        </div>
        <div
          className="side"
        >
          <div
            className="button"
            onClick={[Function]}
          >
            <img
              className="imgButton"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
